docs/pipelines.md

```diff
@@ -574,6 +574,9 @@ After a `Pipeline` has executed the `PipelineRun` will be populated with the `Re
 emitted by the `Pipeline`. These will be written to the `PipelineRun's`
 `status.pipelineResults` field.
 
+If a `Pipeline Result` references a failed `TaskRun` then the `Pipeline Result` is
+considered invalid and will not be emitted by the completed `PipelineRun`.
+
```

Comment:
the result could also be missing because the task was skipped as well...not only when it failed...

Reply:
also successful taskrun without producing task result 🤔

Reply:
> the result could also be missing because the task was skipped as well...not only when it failed...

Ah, good catch, thanks @jerop, updated docs to expand the list of reasons that a pipeline result would be invalid.

Regarding the successful taskrun w/o producing task result, I think that's accounted for - see [my other comment](https://github.com/tektoncd/pipeline/pull/3472#issuecomment-719506324).